models/Review.js

```diff
@@ -67,7 +68,8 @@ ReviewSchema.statics.calculateAverageRating = async function (productId) {
 //post save hook called- on save()
 ReviewSchema.post('save', async function () {
   // we are calling the static method calculateAverageRating that schema has:
-  await this.constructor.calculateAverageRating(this.product); //with this. we access actual schema
+  // AKOSREVIEW: If you define the model before epxorting, then you could also do Review.calculateAverageRating
+  await this.constructor.calculateAverageRating(this.product); //with this. we access actual schema // AKOSREVIEW this == this instance
```

Comment:
calling `this.constructor` works; but another way to do this would be to remove line `80` where the module exports; and instead having something like:

```javascript
const ReviewModel = mongoose.model('Review', ReviewSchema);

ReviewSchema.post('save', async function () {
  await ReviewModel.calculateAverageRating(this.product); //with this. we access actual schema // AKOSREVIEW this == this instance
});


ReviewSchema.post('remove', async function () {
  await ReviewModel.calculateAverageRating(this.product); // we are pointing product id in schema, called product- see line 26
});

module.exports = ReviewModel
```

Reply:
@akosasante thank you so much for showing different approaches that looks much more familiar !! Really appreciate it!

Reply:
No problem. I don't want to get too into depth here since this gets into how Javascript prototype inheritance works, which is good to know but not too relevant to the work you're doing currently; but `this.constructor` is basically doing the same thing because:

*   `this` is referring to the "instance" of the review. So a specific review object/document/result from Mongo
*   `.constructor` is referring to "what class / type of object is this instance?". Which in this case is a `ReviewModel` object. 

So we can skip the indirection if we just already have a variable that refers to that `ReviewModel` type of object.